UnicornRender/video/include/unicorn/video/Transform.hpp

```diff
@@ -0,0 +1,214 @@
+
+namespace unicorn
+{
+namespace video
+{
+/**
+ * @brief Base class for every object that can be translated or rotated
+ */
+class Transform
+{
+public:
+    /** @brief Default constructor */
+    Transform();
+
+    /** @brief Default destructor */
+    virtual ~Transform() = default;
+
+    /**
+     * @brief Sets the direction of the view
+     *
+     * @param[in] direction new direction vectorr
+     */
+    UNICORN_EXPORT void LookAtDirection(glm::vec3 direction);
+
+    /**
+     * @brief Sets the direction of the view
+     *
+     * @param[in] direction new direction vector
+     * @param[in] upVector new up vector
+     */
+    UNICORN_EXPORT void LookAtDirection(glm::vec3 direction, glm::vec3 upVector);
+
+    /**
+     * @brief Sets up vector
+     *
+     * @param[in] upVector new up vector
+     */
+    UNICORN_EXPORT void SetUp(glm::vec3 upVector);
+
+    /**
+     * @brief      Sets the world coordinates
+     *
+     * @param[in]  x world axis
+     * @param[in]  y world axis
+     * @param[in]  z world axis
+     */
+    UNICORN_EXPORT void SetWorldCoordinates(glm::vec3 x, glm::vec3 y, glm::vec3 z);
+
+    /**
+     * @brief Sets translate
+     *
+     * @param[in]  translate new translate vector
+     */
+    UNICORN_EXPORT void SetTranslation(glm::vec3 translate);
+
+    /**
+     * @brief Returns direction
+     *
+     * @return direction
+     */
+    UNICORN_EXPORT glm::vec3 GetDirection() const;
+
+    /**
+     * @brief Return right vector
+     *
+     * @return right vector
+     */
+    UNICORN_EXPORT glm::vec3 GetRight() const;
+
+    /**
+     * @brief Returns up vector
+     *
+     * @return up vector
+     */
+    UNICORN_EXPORT glm::vec3 GetUp() const;
+
+    /**
+     * @brief Return translate
+     *
+     * @return translate
+     */
+    UNICORN_EXPORT glm::vec3 GetTranslation() const;
+
+    /**
+     * @brief Returns model matrix
+     *
+     * @attention Don't forget to recalculate @sa Calculate before
+     *
+     * @return model matrix
+     */
+    UNICORN_EXPORT glm::mat4 const& GetModelMatrix() const;
+
+    /**
+     * @brief Scales object
+     *
+     * @param[in] scale basis scale factors
+     *
+     * Lineary scales each basis component of the mesh vectors by the given scale factors
+     * {1,1,1} is origin scale, {0.5, 0.5, 2} is x and y diminished twice
+     * and z is twice bigger
+     */
+    UNICORN_EXPORT void Scale(glm::vec3 scale);
+
+    /**
+     * @brief Translates by axises
+     *
+     * @param[in] distance per component distance
+     */
+    UNICORN_EXPORT void TranslateLocal(glm::vec3 distance);
+    /**
+     * @brief Translates by world axises
+     *
+     * @param[in] distance per component distance
+     */
+    UNICORN_EXPORT void TranslateWorld(glm::vec3 distance);
+
+    /**
+     * @brief Rotates by X axis counterclockwise
+     *
+     * @param[in] radians amount of radians
+     */
+    UNICORN_EXPORT void RotateX(float radians);
```

Comment:
I think `RotateX/Y/Z` can be removed since you have `void Rotate(float angleRadians, glm::vec3 axis);` or you can inline them in the header to proxy the call to `Rotate(angle, axis)`